distributed/dashboard/scheduler.py

```diff
@@ -319,6 +322,34 @@ def get(self):
         self.set_header("Content-Type", "text/plain")
 
 
+class EventsHandler(WebSocketHandler):
+    def initialize(self, server=None, extra=None):
+        self.server = server
+        self.extra = extra or {}
+        self.plugin = WebsocketPlugin(self, server)
+        self.server.add_plugin(self.plugin)
+
+    def send(self, name, data):
+        data["name"] = name
+        for k in list(data):
+            # Drop bytes objects for now
+            if isinstance(data[k], bytes):
+                del data[k]
+        self.write_message(json.dumps(data))
```

Comment:
It looks like we can just send a dict and Tornado will understand what to do.

Reply:
Rather than override this method maybe we just send messages like `{"name": "add-worker", "worker": "..."}` and use the vanilla Tornado implementation?  I would prefer avoiding wrapping something like this if we can avoid it.

Reply:
I'm not sure I follow. There is no `send` message in the base class. Do you mean to use `self.write_message` instead?

Reply:
Yes, rather than say 

```python
self.send("foo", {...})
```

Maybe we call instead

```python
self.write_message({"name": "foo", ...})
```

This method does three things:

1.  Insert name (which we can do manually)
2.  del bytes objects (I don't understand the motivation for this yet, but perhaps this is why this method exists)
3.  call json.dumps (this I think is not necessary, write_message does this by default)

Reply:
> Insert name (which we can do manually)

I think this was the original reason why I added the method. I wanted to ensure this was consistent. I guess we could manually do this in every place we call this from. (I may have initially overesitamated how often we would call this).

> del bytes objects (I don't understand the motivation for this yet, but perhaps this is why this method exists)

I'm only deleting these because I'm not using them in my demo dashboard and they can't be json dumped. Which isn't really a great reason 😆. Do you know what would happen if we cut out the json dump code and passed through bytes objects?

Reply:
Ah, I actually didn't realize that bytes weren't json serializable (I just learned this).  

OK, maybe it makes sense to keep this in.

Reply:
```suggestion
        self.write_message(data)
```